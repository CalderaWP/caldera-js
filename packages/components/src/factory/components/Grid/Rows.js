import React, { Fragment } from 'react';
import { Row, rowPropTypes } from './Row';
import PropTypes from 'prop-types';

export const Rows = ({ rows, onChange, onBlur, className }) => (
	<Fragment>
		{rows.map(row => (
			<Row
				key={row.rowId}
				{...row}
				onChange={onChange}
				onBlur={onBlur}
				className={className}
			/>
		))}
	</Fragment>
);

Rows.propTypes = {
	rows: PropTypes.arrayOf(PropTypes.shape(rowPropTypes)),
	onChange: PropTypes.func.isRequired,
	onBlur: PropTypes.func,
	className: PropTypes.string
};
