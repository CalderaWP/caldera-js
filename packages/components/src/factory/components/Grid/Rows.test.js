import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	emailField
} from '../../fields.fixtures';

import { Rows } from './Rows';

describe('Row component', () => {
	let onChange = jest.fn();
	let onBlur;
	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});
	const row1 = {
		rowId: 'r1',
		onChange,
		onBlur,
		columns: [
			{
				columnId: 'a1',
				fields: [textField, checkboxField],
				width: '1/2'
			},
			{
				columnId: 'a2',
				fields: [checkboxFieldset],
				width: '1/2'
			}
		]
	};

	const row2 = {
		rowId: 'r2',
		onChange,
		onBlur,
		columns: [
			{
				columnId: 'b1',
				fields: [selectField],
				width: '1/3'
			},
			{
				columnId: 'b2',
				fields: [emailField],
				width: '1/3'
			},
			{
				columnId: 'b3',
				fields: [checkboxField],
				width: '1/3'
			}
		]
	};

	it('Creates with two rows', () => {
		const component = renderer.create(
			<Rows rows={[row1, row2]} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
