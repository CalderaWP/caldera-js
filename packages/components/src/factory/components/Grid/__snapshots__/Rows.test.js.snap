// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component Creates with two rows 1`] = `
Array [
  <div
    className="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
    id="r1"
  >
    <div
      className="caldera-column width-12 sc-bdVaJa gfmZBt"
      width="1/2"
    />
    <div
      className="caldera-column width-12 sc-bdVaJa gfmZBt"
      width="1/2"
    />
  </div>,
  <div
    className="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
    id="r2"
  >
    <div
      className="caldera-column width-12 sc-bdVaJa dbyxub"
      width="1/3"
    />
    <div
      className="caldera-column width-12 sc-bdVaJa dbyxub"
      width="1/3"
    />
    <div
      className="caldera-column width-12 sc-bdVaJa dbyxub"
      width="1/3"
    />
  </div>,
]
`;
